test/Interop/Cxx/class/closure-thunk.swift

```diff
@@ -0,0 +1,21 @@
+// RUN: %target-swiftxx-frontend -I %S/Inputs -emit-silgen %s | %FileCheck %s
```

Comment:
Could you please also add an execution test? That would help to make sure we don't crash in IRGen or at runtime.

Reply:
I can add an execution test, but IRGen will still be incorrect until we fix https://github.com/apple/swift/pull/73019.

Notice that the parameter of the thunk in the test case is `@in NonTrivial`, which means the destructor is called in the thunk. Calling `cfunc2` would result in a double-free crash if `NonTrivial` was defined as follows:

```
struct NonTrivial {
  NonTrivial() { p = new int(123); }
  ~NonTrivial() { delete p; }
  int p = nullptr;
};

void cfunc2(void (*fp)(NonTrivial)) {
  NonTrivial t;
  (*fp)(t);
}
```

Reply:
I see, thank you!